shiny/ui/_toast.py

```diff
@@ -0,0 +1,585 @@
+                    aria_label="Close",
+                )
+            )
+
+        return div({"class": "toast-header"}, *contents, **self.attribs)
+
+
+def toast_header(
+    title: TagChild,
+    *args: TagChild | TagAttrs,
+    icon: Optional[TagChild] = None,
+    status: Optional[str] = None,
+    **kwargs: TagAttrValue,
+) -> ToastHeader:
+    """
+    Create a structured toast header.
+
+    Parameters
+    ----------
+    title
+        Header title text or HTML element.
+    *args
+        Additional content to append to title.
+    icon
+        Optional icon element.
+    status
+        Optional status text (appears muted/right-aligned, e.g., "just now").
+    **kwargs
+        Additional HTML attributes for the header element.
+
+    Returns
+    -------
+    :
+        A toast header object.
+
+    Examples
+    --------
+
+    See :func:`~shiny.ui.toast` for a complete example, including a toast with a
+    header.
+
+    See Also
+    --------
+    * :func:`~shiny.ui.toast`
+    * :func:`~shiny.ui.show_toast`
+    """
+    # Use consolidate_attrs to properly handle TagChild | TagAttrs
+    attrs, children = consolidate_attrs(*args, **kwargs)
+
+    # Combine title with additional children
+    if children:
+        title = TagList(title, *children)
+
+    return ToastHeader(
+        title=title,
+        icon=icon,
+        status=status,
+        attribs=attrs,
+    )
+
+
+# ==============================================================================
+# Show/Hide Functions
+# ==============================================================================
+
+
+def show_toast(
+    toast: str | Toast,
+    *,
+    session: Optional[Session] = None,
+) -> str:
+    """
+    Display a toast notification.
+
+    Toast notifications are temporary, non-intrusive messages that appear on screen.
+    They can be used to provide feedback about actions, display brief messages, or
+    show status updates without interrupting the user's workflow.
+
+    Parameters
+    ----------
+    toast
+        A toast object from :func:`~shiny.ui.toast`, or a plain string (auto-converted to toast).
+    session
+        Shiny session object. If not provided, the session is inferred via
+        :func:`~shiny.session.get_current_session`.
+
+    Returns
+    -------
+    :
+        The toast ID (for use with :func:`~shiny.ui.hide_toast`), or an empty string if
+        the toast has no content (and no action is taken).
+
+    Examples
+    --------
+
+    See :func:`~shiny.ui.toast` for a complete example, including showing and hiding
+    toast notifications.
+
+    See Also
+    --------
+    * :func:`~shiny.ui.toast`
+    * :func:`~shiny.ui.hide_toast`
+    * :func:`~shiny.ui.toast_header`
+    """
+    session = require_active_session(session)
+
+    # Convert string to toast if needed
+    the_toast: Toast
+    if isinstance(toast, Toast):
+        the_toast = toast
+    else:
+        the_toast = Toast(TagList(*toast))
```

Comment:
It's not clear to me why `*toast` (which is a `str`) is used here?

Reply:
ace163abab3b4d7c011b667ec49aa3dca5e262c2